pkg/compactor/compactor.go

```diff
@@ -640,7 +648,8 @@ func (c *Compactor) compactUsers(ctx context.Context) {
 	for _, userID := range users {
 		// Ensure the context has not been canceled (ie. compactor shutdown has been triggered).
 		if ctx.Err() != nil {
```

Comment:
Should we also add the `ctx.Err() == context.Canceled` check here? Or it's not needed?

Reply:
It's not really clear whether we would benefit from a stricter check here, so this is a case where it's probably just best to not touch code we do not need to touch :)